tests_e2e/playwright/navigator.py

```diff
@@ -0,0 +1,448 @@
+            record_video_dir=self.video_dir,
+        )
+        self.page = self.context.new_page()
+        self.initialized = True
+
+    def teardown(self) -> None:
+        """Shut down and close playwright. This is important to ensure that
+        no leftover processes are left running in the background."""
+        self.context.close()
+        self.browser.close()  # Make sure to close, so that videos are saved.
+        self.playwright.stop()
+        logger.debug(">>> Teardown complete.")
+
+    def login(self) -> None:
+        """Login to nebari deployment using the auth method on the class."""
+        if self.auth == "google":
+            self.login_google()
+        elif self.auth == "password":
+            self.login_password()
+        else:
+            raise ValueError(f"Auth type of {self.auth} is invalid.")
+
+    def login_google(self) -> None:
+        """Go to a nebari deployment, login via Google"""
+        logger.debug(">>> Sign in via Google and start the server")
+        self.page.goto(self.nebari_url)
+        expect(self.page).to_have_url(re.compile(f"{self.nebari_url}*"))
+
+        self.page.get_by_role("button", name="Sign in with Keycloak").click()
+        self.page.get_by_role("link", name="Google").click()
+        self.page.get_by_role("textbox", name="Email or phone").fill(self.username)
+        self.page.get_by_role("button", name="Next").click()
+        self.page.get_by_role("textbox", name="Enter your password").fill(self.password)
+
+        self.page.wait_for_load_state("networkidle")
+        self.page.get_by_role("button", name="Next").click()
+
+        # let the page load
+        self.page.wait_for_load_state("networkidle")
+
+    def login_password(self) -> None:
+        """Go to a nebari deployment, login via Username/Password, and start
+        a new server.
+        """
+        logger.debug(">>> Sign in via Username/Password")
+        self.page.goto(self.nebari_url)
+        expect(self.page).to_have_url(re.compile(f"{self.nebari_url}*"))
+
+        self.page.get_by_role("button", name="Sign in with Keycloak").click()
+        self.page.get_by_label("Username").fill(self.username)
+        self.page.get_by_label("Password").click()
+        self.page.get_by_label("Password").fill(self.password)
+        self.page.get_by_role("button", name="Sign In").click()
+
+        # let the page load
+        self.page.wait_for_load_state("networkidle")
+
+    def start_server(self) -> None:
+        """Start a nebari server. There are several different web interfaces
+        possible in this process depending on if you already have a server
+        running or not. In order for this to work, we look for html elements,
+        wait because the page may need to load, and if they don't appear, we
+        check for another server start option.
+        """
+        # if server is not yet running
+        start_locator = self.page.get_by_role("button", name="Start My Server")
+        try:
```

Comment:
Instead of trying to click a button that might not be there, can't we just check that directly? We could just check if `start_locator` is visible, no? If not, we can then try the other button.